### PR TITLE
amending bump version to prevent duplicate version commits

### PR DESCRIPTION
**Pull request by slemasne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: ci_tools/bumpversion.sh [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
-git config --local user.email "engineering@finbourne.com"
-git config --local user.name "finbourne-bot-public"
-git push https://finbourne-bot-public:${gh_token}@github.com/finbourne/lusid-python-tools.git HEAD:$TRAVIS_BRANCH
-echo "Making commit with incremented version"
\ No newline at end of file
+#!/bin/bash
+
+last_commit_message=$(git log -1 HEAD --pretty=format:%s)
+
+echo "Last commit message: $last_commit_message"
```
Comment (rizwansaeed) [inline ci_tools/bumpversion.sh:5]: this looks like debug, I don't think we need to print it out

File: ci_tools/bumpversion.sh [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
+if [[ "$last_commit_message" == "Bump version"* ]]; then
+        echo "last commit is a bump version commit... skipping..."
```
Comment (rizwansaeed) [inline ci_tools/bumpversion.sh:8]: The pedant in me dislikes the inconsistent casing in the log messages, sorry!

I think change to something like "Skipping version increment (no commits found)"

File: ci_tools/bumpversion.sh [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+else
+        echo "Bumping packaging patch version by one"
```
Comment (rizwansaeed) [inline ci_tools/bumpversion.sh:10]: Maybe just a simple "Incrementing patch version to $new_version"

File: ci_tools/bumpversion.sh (hunk 1 above)
Comment (slemasne) [reply ci_tools/bumpversion.sh:5]: Sure, removed.

File: ci_tools/bumpversion.sh (hunk 2 above)
Comment (slemasne) [reply ci_tools/bumpversion.sh]: Sure. done.

File: ci_tools/bumpversion.sh (hunk 3 above)
Comment (slemasne) [reply ci_tools/bumpversion.sh:8]: I'm not sure the Travis environment will know the $new_version variable? We could always scrape this from PyPi?

Comment (rizwansaeed) [reply ci_tools/bumpversion.sh:8]: if we don't have it at hadn't then I think its ok to leave

Comment (slemasne) [reply ci_tools/bumpversion.sh]: See latest commit. We can take the version from `__version__.py` and then increment in the shell script.

File: ci_tools/bumpversion.sh [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
-git push https://finbourne-bot-public:${gh_token}@github.com/finbourne/lusid-python-tools.git HEAD:$TRAVIS_BRANCH
-echo "Making commit with incremented version"
\ No newline at end of file
+#!/bin/bash
+
+last_commit_message=$(git log -1 HEAD --pretty=format:%s)
+
+# Find next version
+
+api_version=$(cat __version__.py | grep __version__ |  awk '{split($0, a, "="); print a[2]}' | tr -d ' "')
```
Comment (rizwansaeed) [inline ci_tools/bumpversion.sh:7]: i think this block can be moved to line 18 (where it is used)

Comment (slemasne) [reply ci_tools/bumpversion.sh]: Done

**Pull request merged**